dashboard/app/jobs.go

```diff
@@ -455,6 +546,82 @@ func createJobResp(c context.Context, job *Job, jobKey *db.Key) (*dashapi.JobPol
 	return resp, false, nil
 }
 
+// It would be easier to just check if the User field is empty, but let's also not
+// miss the situation when some actual user sends a patch testing request without
+// patch.
+func isRetestReproJob(job *Job, build *Build) bool {
+	if job.Type != JobTestPatch || job.Patch != 0 {
```

Comment:
Positive conditions are usually easier to understand and verify than defining the predicate as a negation of all negative conditions:
```
	return job.Type == JobTestPatch && job.Patch == 0 &&
		job.KernelRepo == build.KernelRepo && job.KernelBranch == build.KernelBranch
```